src/providers/krb5/krb5_utils.c

```diff
@@ -570,6 +573,18 @@ parse_krb5_map_user(TALLOC_CTX *mem_ctx, const char *krb5_map_user,
         }
     }
 
+    /* conversion names to fully-qualified names */
+    for (int i = 0; i < size; i++) {
+        fq_name = sss_create_internal_fqname(tmp_ctx,
+                                             name_to_primary[i].id_name,
+                                             dom_name);
+        name_to_primary[i].id_name = talloc_strdup(name_to_primary, fq_name);
```

Comment:
Both these calls should have their return value checked. Unless you're using strdup to get around a const-issue, you can also directly assign to id_name.